stages/org.osbuild.ostree.genkey

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python3
+"""
+Generate ed25519 public/private keypair in format used by ostree.
+"""
+
+import os
+import subprocess
+import sys
+import tempfile
+
+import osbuild.api
+
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "required": ["publickey", "secretkey"],
+  "properties": {
+    "publickey": {
+      "description": "Path of generated public key",
+      "type": "string"
+    },
+    "secretkey": {
+      "description": "Path of generated secret key",
+      "type": "string"
+    }
+  }
+}
+"""
+
+
+def openssl(*args):
+    args = list(args)
+    print("openssl " + " ".join(args), file=sys.stderr)
+    subprocess.run(["openssl"] + args,
+                   encoding="utf8",
+                   stdout=sys.stderr,
+                   input=None,
+                   check=True)
+
+
+def openssl_stdout(*args):
+    args = list(args)
+    print("openssl " + " ".join(args), file=sys.stderr)
+    res = subprocess.run(["openssl"] + args,
+                         stdout=subprocess.PIPE,
+                         input=None,
+                         check=True)
+
+    return res.stdout
+
+# Based on gen_ed25519_keys() in https://github.com/ostreedev/ostree/blob/main/tests/libtest.sh
+
+
+def main(args, options):
+    tree = args["tree"]
+    pubkeyfile = os.path.join(tree, options["publickey"].lstrip("/"))
+    seckeyfile = os.path.join(tree, options["secretkey"].lstrip("/"))
+
+#    # Openssl default config is in /etc which is overwritten, so use an empty config
+#    with open('/etc/openssl.conf', 'w') as f:
+#       f.write('\n')
+
+    with tempfile.TemporaryDirectory(dir=tree) as tmpdir:
```

Comment:
Since we really really want the private key to be thrown away, it probably makes sense to look at something stronger than `TemporaryDirectory`?  Maybe have openssl write to a pipe and we retain it just in memory, and then explicitly drop it?

Reply:
The resulting secure key file is written to disk by the stage anyway, as it is then later used by the org.osbuild.ostree.sign stage. I don't see a way to pass files between stages without hitting disk anyway. However, these files are only temporarily stored on the build box, they don't get stored in any produced output.